src/features/posts.ts

```diff
@@ -0,0 +1,83 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getPosts, getUserPosts } from '../api/posts';
+import { Post } from '../types/Post';
+
+interface PostsState {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: PostsState = {
+  items: [] as Post[],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchPosts = createAsyncThunk('posts/fetchPosts', async () => {
+  const posts = await getPosts();
+
+  return posts;
+});
+
+export const fetchUserPosts = createAsyncThunk(
+  'posts/fetchuserPosts',
```

Comment:
There is a minor inconsistency in the action type name `fetchuserPosts`. It should be `fetchUserPosts` to maintain consistency and clarity in naming conventions.